### PR TITLE
Added fix for exporting Waters methods containing compounds with multiple modifications

### PR DESCRIPTION
**Pull request by kaipot**

### CODE REVIEW

File: pwiz_tools/Skyline/Model/Export.cs [hunk 1]
```diff
@@ -4125,6 +4125,33 @@ protected override void WriteHeaders(TextWriter writer)
             }
             writer.WriteLine();
         }
+
+        // Hack to replace modification brackets with parentheses.
```
Comment (brendanx67) [inline pwiz_tools/Skyline/Model/Export.cs:4129]: Make this a block comment.

File: pwiz_tools/Skyline/Model/Export.cs [hunk 2]
```diff
@@ -4125,6 +4125,33 @@ protected override void WriteHeaders(TextWriter writer)
...
+
+        // Hack to replace modification brackets with parentheses.
+        // MassLynx or VerifyESkylineLibrary.dll has some problem with multiple sets of brackets that causes
+        // an incomplete method to be generated.
+        public static string FormatMods(string modSeq)
+        {
+            if (!modSeq.Contains('['))
+                return modSeq;
+
+            var newSeq = new StringBuilder(modSeq.Length);
```
Comment (brendanx67) [inline pwiz_tools/Skyline/Model/Export.cs:4137]: I wonder if this is really necessary. It seems like replacing this entire function with:

return modeSeq.Replace('[', '(').Replace(']', ')');

Would be just fine given all the other work that is happening. I can't imagine the double allocation ever showing up in a profiler as important.

**Review by brendanx67 — CHANGES_REQUESTED** [review]
Thanks for fixing this so quickly.

File: pwiz_tools/Skyline/Model/Export.cs [hunk 3]
```diff
@@ -4125,6 +4125,15 @@ protected override void WriteHeaders(TextWriter writer)
             }
             writer.WriteLine();
         }
+
+        /* Hack to replace modification brackets with parentheses.
```
Comment (brendanx67) [inline pwiz_tools/Skyline/Model/Export.cs:4129]: Sorry, I meant our standard C# function header comment style:
    /// <summary>
    /// text
    /// </summary>

**Review by brendanx67 — APPROVED** [review]

**Comment by chambm** [comment]
OK I think this is good for cherry picking.

**Pull request merged**